cydarm_connector.py

```diff
@@ -0,0 +1,347 @@
+        func = self.cydarm.add_member_to_case
+        kwargs = self.extract_args_dict(param, ["case_uuid", "member_case_uuid"])
+        return self.call_cydarm_api(func, kwargs)
+
+    def _handle_add_case_tag(self, param):
+        func = self.cydarm.add_case_tag
+        kwargs = self.extract_args_dict(param, ["case_uuid", "tag_value"])
+        return self.call_cydarm_api(func, kwargs)
+
+    def call_cydarm_api(self, cydarm_api_func, kwargs):
+        self.save_progress(f"Calling {cydarm_api_func.__name__} with kwargs: {kwargs}")
+        result = cydarm_api_func(**kwargs)
+        self.save_progress(f"Output from {cydarm_api_func.__name__}: {result}")
+        return result
+
+    def _handle_delete_case_tag(self, param):
+        func = self.cydarm.delete_case_tag
+        kwargs = self.extract_args_dict(param, ["case_uuid", "tag_value"])
+        return self.call_cydarm_api(func, kwargs)
+
+    def generate_action_result(self, param, request_func: Callable):
+        self.save_progress("In action handler for: {0}".format(self.get_action_identifier()))
+
+        action_result = self.add_action_result(ActionResult(dict(param)))
+
+        response = request_func(param)
+        num_items = 1
+        if type(response) == list:
+            num_items = len(response)
+            for item in response:
+                action_result.add_data(item)
+        elif type(response) == dict:
+            action_result.add_data(response)
+        else:
+            self.save_progress("No response data.")
+
+        action_result.update_summary({
+            "total_objects": num_items,
+            "total_objects_successful": num_items,
+        })
+
+        return action_result.set_status(phantom.APP_SUCCESS)
+
+    def handle_action(self, param):
+        # Get the action that we are supposed to execute for this App Run
+        action_id = self.get_action_identifier()
+
+        self.debug_print("action_id", self.get_action_identifier())
+
+        if action_id == 'test_connectivity':
+            return self._handle_test_connectivity(param)
+        else:
+            func_name = f"_handle_{action_id}"
+            func = getattr(self, func_name, None)
+            if func is None:
+                raise RuntimeError(f"No function found called: {func_name}")
+            return self.generate_action_result(param, func)
+
+    def initialize(self):
+        # Load the state in initialize, use it to store data
+        # that needs to be accessed across actions
+        self._state = self.load_state()
+
+        # get the asset config
+        config = self.get_config()
+        """
+        # Access values in asset config by the name
+
+        # Required values can be accessed directly
+        required_config_name = config['required_config_name']
+
+        # Optional values should use the .get() function
+        optional_config_name = config.get('optional_config_name')
+        """
+
+        def log_function(string):
```

Comment:
Why do this and why do this here? You can just pass `self.save_progress` to `CydarmAPI` constructor

Reply:
fixed